tests/tape/test_qnode.py

```diff
@@ -368,6 +368,44 @@ def func():
         # No differentiation required. No error raised.
         grad()
 
+    def test_unrecognized_keyword_arguments_validation(self):
+        """Tests that a UserWarning is raised when unrecognized keyword arguments are provided."""
+
+        # use two unrecognized methods, to confirm that multiple warnings are raised
+        unrecognized_one = "test_method_one"
+        unrecognized_two = "test_method_two"
+        warning_text = (
+            " is unrecognized, and will not be included in your computation. "
+            "Please review the QNode class or qnode decorator for the list of available "
+            "keyword variables."
+        )
+
+        expected_warnings = {
+            (UserWarning, unrecognized_one + warning_text),
+            (UserWarning, unrecognized_two + warning_text),
+        }
+
+        dev = qml.device("default.qubit", wires=1, shots=1)
+
+        # test with QNode
+        with pytest.warns(UserWarning) as warning_list:
+            QNode(dummyfunc, dev, test_method_one=1, test_method_two=2)
+
+        warnings = {(warning.category, warning.message.args[0]) for warning in warning_list}
+        assert warnings == expected_warnings
+
+        # test with QNode decorator
+        with pytest.warns(UserWarning) as decorator_warning_list:
+
+            @qml.qnode(dev, test_method_one=1, test_method_two=2)
+            def circ():
+                return qml.expval(qml.PauliZ(0))
+
+        warnings = {
+            (warning.category, warning.message.args[0]) for warning in decorator_warning_list
+        }
+        assert warnings == expected_warnings

```

Comment:
This could be put in a separate test case (either as a separate test function or by using parametrization) for slightly improve clarity.

Reply:
@thisac I've added some parameterization to the test which should hopefully improve clarity, and I've included all your recommendations above.